### DISCUSSION TITLE
Create the files `README.md` and `defaults/main.yml` from `meta/argument_specs.yml`

### DISCUSSION THREAD

**Discussion by berndfinger** [sap-linuxlab/community.sap_install#538] [Show and tell]
By slightly limiting the variety of entries in `meta/argument_specs.yml`, we can reliably create the variable section of a role's `README.md` and the whole file `defaults/main.yml` automatically. I published an initial version here: https://github.com/berndfinger/tool.sap_shell_scripts/tree/create_role_files_from_argument_specs/create_role_files_from_argument_specs. The README.md file in this directory should have all the required information for using the tool.

Please let me know what you think about it.

**Comment by Wabri** [comment]
Good idea, why not open a new PR?

**Reply by sean-freeman** [reply]
Only so many hours in a day, and functionality always win in prioritisation :slightly_smiling_face: This is already partially implemented in a few roles but it is low priority to finish in all other Ansible Roles - see [search](https://github.com/search?q=repo%3Asap-linuxlab%2Fcommunity.sap_install+%22Role+Input+Parameters%22+language%3AMarkdown&type=code)
